### DISCUSSION TITLE
Option for loo

### DISCUSSION THREAD

**Discussion by mattocci27** [wlandau/stantargets#61] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

The addition of an option like `return_loo = TRUE`, similar to `return_summary = TRUE` might be a good idea because it is hard to compute `loo::loo` when `tar_stan_mcmc` is run inside `tar_map`.

For example, we can use `summary` inside this `tar_map` like:

```
mapped <- tar_map(
  values = list(data = 1:5),
  tar_stan_mcmc(
      fit,
      "model.stan",
      ...),
  tar_targets(
     summary_modified,
     some_functions(fit_summary)
  ))
```

However, to use `loo::loo()` for each output, we need another `tar_map`, which can be something like:

```
loo_mapped <- tar_map(
  values = list(mcmc = rlang::syms(paste0("fit_mcmc_model_, 1:5))),
  tar_target(
    loo,
    loo_function(mcmc)
  )
)
```

Considering that `tar_map` is useful for creating different parameter settings, it makes sense that `stantargets` should be able to handle `loo` computations for model comparisons. What do you think?

**Comment by wlandau** [comment]
Definitely a neat idea. However, I hesitate to directly build in features this specific, since the Stan ecosystem changes rapidly and there are many different ways to invoke `loo`.
